### PR TITLE
s/autopilot/autoupdate/ mostly (leave the unit files alone). Also update the documentation.

### PR DESCRIPTION
**Pull request by chipaca**
This fixes issue #37.

### CODE REVIEW

File: cmd/snappy/cmd_update.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ const (
 	shutdownTimeout = "+10"
 )
 
-var shutdownMsg = i18n.G("snappy autopilot triggered a reboot to boot into an up to date system -- temprorarily disable the reboot by running 'sudo shutdown -c'")
```
Comment (chipaca) [inline cmd/snappy/cmd_update.go]: Note I'd much rather change this to “snappy autoupdate needs to reboot to finish an update. Cancel with 'shutdown -c'” which (I think) communicates the same thing, but keeps it under 80 cols. Wall doesn't word-wrap, so the longer it is the more chances of it being untidy.

File: coreconfig/config.go [hunk 2]
```diff
@@ -73,13 +73,13 @@ var (
...
 
 type systemConfig struct {
-	Autopilot *bool           `yaml:"autopilot,omitempty"`
-	Timezone  *string         `yaml:"timezone,omitempty"`
-	Hostname  *string         `yaml:"hostname,omitempty"`
-	Modprobe  *string         `yaml:"modprobe,omitempty"`
-	Modules   []string        `yaml:"load-kernel-modules,omitempty"`
-	Network   *networkConfig  `yaml:"network,omitempty"`
-	Watchdog  *watchdogConfig `yaml:"watchdog,omitempty"`
+	AutoUpdate *bool           `yaml:"autoupdate,omitempty"`
```
Comment (chipaca) [inline coreconfig/config.go]: backwards compatibility could be preserved here, but I'm pretty sure it's not worth it, given that (as it turns out) we hadn't actually documented this usage.

File: docs/autoupdate.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+reboot, although a message is printed to console with instructions on
+how to abort the reboot, in case you are logged in at the time.
+
+## Implementation details
+
+Autoupdate used to be called *autopilot* (but that got very confusing,
+especially when people were using snappy with other things that have
+their own autopilot, like an OpenStack deployment that used
+Canonical's own OpenStack Autopilot, or in mobile robots that could
+fly themselves); the `systemd` units still use this name.
```
Comment (chipaca) [inline docs/autoupdate.md]: we could also rename the units, but I'm not at all sure it's worth it; rollback would be very tricky, and it's not really user-visible.

File: cmd/snappy/cmd_update.go (hunk 1 above)
Comment (mvo5) [inline cmd/snappy/cmd_update.go]: +1 for this

File: coreconfig/config.go [hunk 4]
```diff
@@ -48,9 +48,9 @@ var (
 )
 
 const (
-	autopilotTimer         string = "snappy-autopilot.timer"
-	autopilotTimerEnabled  string = "enabled"
-	autopilotTimerDisabled string = "disabled"
+	autoUpdateTimer         string = "snappy-autopilot.timer"
```
Comment (mvo5) [inline coreconfig/config.go]: Should we disable the file as well? i.e. snappy-autopilot.timer -> snappy-autoupdate.timer ?

File: docs/autoupdate.md (hunk 3 above)
Comment (mvo5) [inline docs/autoupdate.md]: Hm, good point

**Comment by mvo5** [comment]
I'm in favor of this change. I think the new name is clearer than the previous one.

File: cmd/snappy/cmd_update.go (hunk 1 above)
Comment (niemeyer) [inline cmd/snappy/cmd_update.go]: Are you going to update the branch to bring this in, John, or is your intention to do that later?

File: _integration-tests/tests/autoupdate-msg_test.go [hunk 5]
```diff
@@ -50,7 +50,7 @@ func (s *autopilotMsgSuite) TestAutoPilotMessageIsPrinted(c *check.C) {
 
 	expected := "(?ms)" +
 		".*" +
-		"^The snappy autopilot is updating your system.*\n" +
+		"^snappy autoupdate is updating your system.*\n" +
```
Comment (niemeyer) [inline _integration-tests/tests/autoupdate-msg_test.go]: Do we even need a term for this?  Why isn't that saying "snappy is automatically updating your system"?

File: cmd/snappy/common.go [hunk 6]
```diff
@@ -52,12 +52,12 @@ func withMutexAndRetry(f func() error) error {
...
 		if err == priv.ErrAlreadyLocked {
 			var msg string
-			if isAutoPilotRunning() {
+			if isAutoUpdateRunning() {
 				// FIXME: we could even do a
 				//    journalctl -u snappy-autopilot
 				// here
 				msg = i18n.G(
-					`The snappy autopilot is updating your system in the background. This may
+					`snappy autoupdate is updating your system in the background. This may
```
Comment (niemeyer) [inline cmd/snappy/common.go]: Same.. why do we need to call out "snappy autoupdate" as an entity? Why doesn't that say "snappy is updating your system in the background."?

File: cmd/snappy/cmd_update.go (hunk 1 above)
Comment (chipaca) [inline cmd/snappy/cmd_update.go]: yes, i'll do it now that it seems we're all in agreement

File: cmd/snappy/common.go (hunk 6 above)
Comment (chipaca) [inline cmd/snappy/common.go]: Good point.

File: docs/autoupdate.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+## Usage
+
+To check whether the feature is active, run
+
+    snappy config ubuntu-core | grep autoupdate
+
+If you want to disable it run
+
+    echo 'config: {ubuntu-core: {autoupdate: off}}' | sudo snappy config ubuntu-core
```
Comment (niemeyer) [inline docs/autoupdate.md]: Eventually we'll need something more friendly for this. Something along the lines of:

    sudo snappy-config ubuntu-core config.ubuntu-core.autoupdate=off

**Comment by niemeyer** [comment]
LGTM. I would just prefer to have the autoupdate-the-entity removed.

**Comment by chipaca** [comment]
could you expand that? you'd rather snappy systems didn't autoupdate at all?

File: docs/autoupdate.md (hunk 7 above)
Comment (chipaca) [inline docs/autoupdate.md]: Yes. I've bee in favour of this since forever.

File: _integration-tests/tests/autoupdate-msg_test.go (hunk 5 above)
Comment (chipaca) [inline _integration-tests/tests/autoupdate-msg_test.go]: Yeh, i'll drop it entirely.

**Comment by niemeyer** [comment]
No, sorry, it was a reference to the earlier comment, which you seemed to agree with.

File: cmd/snappy/cmd_update.go (hunk 1 above)
Comment (chipaca) [inline cmd/snappy/cmd_update.go]: done, if you want to once-over one last time

File: cmd/snappy/cmd_update.go [hunk 8]
```diff
@@ -53,7 +53,8 @@ const (
...
+// TRANSLATORS: please keep this under 80 characters if possible
+var shutdownMsg = i18n.G("Snappy needs to reboot to finish an update. Defer with 'shutdown -c'.")
```
Comment (mvo5) [inline cmd/snappy/cmd_update.go]: Hm, minor nitpick, "sudo shutdown -c" was used in the old one because some people tripped over it.

**Pull request merged**